### PR TITLE
Add hmac verification on ipn

### PR DESCRIPTION
**Pull request by joyet-simon**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-1822/[-sylius]-add-hmac-verification-on-ipn)
 Need to implement HMAC verification for IPN

### Code changes

We add a SecurityHelper, a new AlmaException and a SecurityException.
In the SecurityHelper, we use the php client's function to make a verification on the signature.
We call this helper's function in the notify action (=IPN) and we throw exception when it's necessary.  

### How to test

_As a reviewer, you are encouraged to test the PR locally._

Try to make a payment on the [QA ENV](https://hmac-qa-2-sylius-1-11-4.integrations.getalma.eu/fr_FR/), leave the Alma checkout before the return on the success page to get IPN call. Check on Datadog if everything is fine.
You can use POSTMAN to make a call with no signature or a bad signature.

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: .github/workflows/aqua.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Aqua
```
Comment (FranceBe) [inline .github/workflows/aqua.yml]: this does not look like part of the initial ticket ? is there a reason to include those change in this PR ? :o

Comment (joyet-simon) [reply .github/workflows/aqua.yml]: Yes, I was just because I make a mistake on the base branch for the pr

File: src/Helper/SecurityHelper.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	/**
+	 * @param string $paymentId
+	 * @param string $key
+	 * @param string $signature
+	 * @throws SecurityException
+	 */
+	public function isHmacValidated(string $paymentId, string $key, string $signature): void
+	{
+		 if (!$this->paymentValidator->isHmacValidated($paymentId, $key, $signature)) {
+			 throw new SecurityException("HMAC validation failed for payment $paymentId");
```
Comment (Benjamin-Freoua-Alma) [inline src/Helper/SecurityHelper.php]: I think we can add the api_key and signature in the message throwed. WDYT ?

File: src/Payum/Action/NotifyAction.php [hunk 3]
```diff
@@ -19,78 +20,120 @@
...
+		$query = ArrayObject::ensureArrayObject($httpRequest->query);
+
+		/** @var string $payment_id */
+		$payment_id = $query->get(AlmaBridgeInterface::QUERY_PARAM_PID);
+
+		/** @var PaymentInterface $payment */
+		$payment = $request->getFirstModel();
+
+		try {
+			$this->securityHelper->isHmacValidated(
```
Comment (Benjamin-Freoua-Alma) [inline src/Payum/Action/NotifyAction.php]: The HMAC verification didn't work, I think we need to explore how does work the NotifyAction and tokenFactory

Comment (Benjamin-Freoua-Alma) [reply src/Payum/Action/NotifyAction.php]: Capture d’écran 2024-10-03 à 16 34 44

**Review by Benjamin-Freoua-Alma — CHANGES_REQUESTED** [review]

File: src/Helper/SecurityHelper.php (hunk 2 above)
Comment (joyet-simon) [reply src/Helper/SecurityHelper.php]: Done

File: src/Helper/SecurityHelper.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	/**
+	 * @param string $paymentId
+	 * @param string $key
+	 * @param string $signature
+	 * @throws SecurityException
+	 */
+	public function isHmacValidated(string $paymentId, string $key, string $signature): void
+	{
+		 if (!$this->paymentValidator->isHmacValidated($paymentId, $key, $signature)) {
+			 throw new SecurityException("HMAC validation failed for payment $paymentId - API key: $key, signature: $signature");
```
Comment (Francois-Gomis) [inline src/Helper/SecurityHelper.php]: Warning  : Leak API KEY !!!!

**Review by Francois-Gomis — CHANGES_REQUESTED** [review]

File: src/Helper/SecurityHelper.php (hunk 4 above)
Comment (joyet-simon) [reply src/Helper/SecurityHelper.php]: Done

File: src/Helper/SecurityHelper.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+class SecurityHelper
+{
+	/**
+	 * @var PaymentValidator
+	 */
+	protected $paymentValidator;
+
+	public function __construct()
+	{
+		$this->paymentValidator = new PaymentValidator();
```
Comment (Francois-Gomis) [inline src/Helper/SecurityHelper.php]: Ad Injection dependency, i think autowiring is enable !

**Review by Francois-Gomis — CHANGES_REQUESTED** [review]

**Review by Benjamin-Freoua-Alma — APPROVED** [review]
Without QA Test

**Pull request closed without merging**